File: API/Entities/Blog.cs
namespace API.Entities;

public class Blog
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime CreatedDate { get; set; }
    public Category Category { get; set; }
    // [Column(TypeName = "jsonb")]
    public string Content { get; set; }
    public string ThumbPublicId { get; set; }
    public string ThumbUrl { get; set; }
}

File: client-app/src/app/types/blog.ts
import { Category } from './category';

export interface Blog {
	id: string;
	title: string;
	description: string;
	createdDate: Date;
	category: Category;
	content: string;
	thumbPublicId: string;
	thumbUrl: string;
}

export class Blog implements Blog {
	id = '';
	title = '';
	description = '';
	createdDate = new Date();
	category = new Category();
	content = '';
	thumbPublicId = '';
	thumbUrl = '';

	constructor(blog?: Blog) {
		if (blog) {
			Object.assign(this, blog);
		}
	}
}
